axiom/nr_log_level.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef NR_LOG_LEVEL_HDR
+#define NR_LOG_LEVEL_HDR
+
+#include <stdbool.h>
+
+/*
+ *  Implementation Based on:
+ *      https://www.php-fig.org/psr/psr-3/#5-psrlogloglevel
+ *      https://datatracker.ietf.org/doc/html/rfc5424#section-6.2.1
+ */
+
+#define LOG_LEVEL_EMERGENCY (0)  // system is unusable
+#define LOG_LEVEL_ALERT (1)      // action must be taken immediately
+#define LOG_LEVEL_CRITICAL (2)   // critical conditions
+#define LOG_LEVEL_ERROR (3)      // error conditions
+#define LOG_LEVEL_WARNING (4)    // warning conditions
+#define LOG_LEVEL_NOTICE (5)     // normal but significant conditions
+#define LOG_LEVEL_INFO (6)       // informational messages
+#define LOG_LEVEL_DEBUG (7)      // debug-level messages
+#define LOG_LEVEL_UNKNOWN (8)    // NON PSR- Unknown/Undefined log level
+#define LOG_LEVEL_DEFAULT (LOG_LEVEL_WARNING)
+
+#define LL_EMER_STR "EMERGENCY"
+#define LL_ALER_STR "ALERT"
+#define LL_CRIT_STR "CRITICAL"
+#define LL_ERRO_STR "ERROR"
+#define LL_WARN_STR "WARNING"
+#define LL_NOTI_STR "NOTICE"
+#define LL_INFO_STR "INFO"
+#define LL_DEBU_STR "DEBUG"
+#define LL_UNKN_STR "UNKNOWN"
```

Comment:
A small (not enough to hold back merging) nit about the inconsistent naming convention prefix: `LL` vs `LOG_LEVEL`